bugmon_tc/monitor/monitor.py

```diff
@@ -0,0 +1,138 @@
+        :param dry_run: Boolean indicating if bugs should be updated
+        """
+        self.bugsy = Bugsy(api_key=api_key, bugzilla_url=api_root)
+        self.artifact_dir = artifact_dir
+        self.dry_run = dry_run
+
+    @property
+    def in_taskcluster(self):
+        """
+        Helper to determine if we're in a taskcluster worker
+
+        :return: bool
+        """
+        return "TASK_ID" in os.environ and "TASKCLUSTER_ROOT_URL" in os.environ
+
+    def fetch_bugs(self):
+        """
+        Generate EnhancedBug instances for all actionable bugs
+
+        :return: list of EnhancedBug
+        """
+        response = self.bugsy.request("bug", params=QUERY)
+        bugs = [EnhancedBug(self.bugsy, **bug) for bug in response["bugs"]]
+        for bug in sorted(bugs, key=lambda bug: bug.id):
+            if self.is_actionable(bug):
+                # Ensure that attachments and comments have been retrieved
+                if bug.attachments is None:
+                    bug.get_attachments()
+                if bug.comments is None:
+                    bug.get_comments()
+
+            yield bug
+
+    def is_actionable(self, bug):
+        """
+        Determine which action, if any, can be performed on the bug
+
+        :param bug: Bug to analyse
+        :return: bool
+        """
+        with tempfile.TemporaryDirectory() as temp_dir:
+            try:
+                bugmon = BugMonitor(self.bugsy, bug, temp_dir, self.dry_run)
+                LOG.info(f"Analyzing bug {bug.id} (Status: {bug.status})")
+
+                if bugmon.is_supported():
+                    if bugmon.needs_verify():
+                        LOG.info(f"Queing bug {bug.id} for verification")
+                        return True
+                    if bugmon.needs_confirm():
+                        LOG.info(f"Queing bug {bug.id} for confirmation")
+                        return True
+                    if bugmon.needs_bisect():
+                        LOG.info(f"Queing bug {bug.id} for bisection")
+                        return True
+
+            except BugException as e:
+                LOG.error(f"Error processing bug {bug.id}: {e}")
+
+        return False
+
+    def create_tasks(self):
+        """
+        Fetch all bugs and generate artifacts representing the tasks that need to be
+        performed on those bugs
+        """
+
+        for bug in self.fetch_bugs():
+            bug_uuid = uuid.uuid1()
+            monitor_path = f"monitor-{bug_uuid}.json"
+
+            # Write monitor artifact
+            with open(os.path.join(self.artifact_dir, monitor_path), "w") as file:
+                bug_data = bug.to_json()
+                json.dump(json.loads(bug_data), file, indent=2)
+
+            # This will only be set if we're running in taskcluster
+            parent_id = os.getenv("TASK_ID", "0")
+
+            process_path = f"process-{bug_uuid}.json"
+            report_path = f"reporter-{bug_uuid}.json"
```

Comment:
This is never used, right? Why create it?

Reply:
You're correct this could be optimized out but I'm going to leave it so that I can write the task to disk when used locally.